### PR TITLE
Update platform channels docs for Kotlin & Swift

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/flutter/flutter/issues/11032
Fixes https://github.com/flutter/flutter/issues/10400

### CODE REVIEW

File: platform-channels.md [hunk 1]
```diff
@@ -405,61 +535,36 @@ Objective-C:
...
-    ```
-1. Validate that the project builds (shortcut `cmd-b`).
-
-With the project converted to use Swift for the main application class we can
-now begin.
+navigator.
 
-First we override `didFinishLaunchingWithOptions` to contain a
-`FlutterMethodChannel` tied to the channel name `samples.flutter.io/battery`:
+Next, override extend the `application` function create a `FlutterMethodChannel`
```
Comment (mravn-google) [inline platform-channels.md]: Remove "override" or "extend".

Add "to" or "and" before "create".

The name of the `application` function is really `application(_:didFinishLaunchingWithOptions:)`. There are many functions called `application` in a `UIApplicationDelegate`.

File: platform-channels.md [hunk 2]
```diff
@@ -405,61 +535,36 @@ Objective-C:
...
 @UIApplicationMain
 @objc class AppDelegate: FlutterAppDelegate {
   override func application(
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
-    GeneratedPluginRegistrant.register(with: self)
-    let controller : FlutterViewController = window?.rootViewController as! FlutterViewController;
+    GeneratedPluginRegistrant.register(with: self);
+
+    let controller : FlutterViewController = window?.rootViewController as!FlutterViewController;
```
Comment (mravn-google) [inline platform-channels.md]: No space after `as!`?

File: platform-channels.md [hunk 3]
```diff
@@ -405,61 +535,36 @@ Objective-C:
...
                                                    binaryMessenger: controller);
-  
     batteryChannel.setMethodCallHandler({
       (call: FlutterMethodCall, result: FlutterResult) -> Void in
       // Handle battery messages.
     });
-   
-    return true
+
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions);
```
Comment (mravn-google) [inline platform-channels.md]: We forgot to return `[super application:application didFinishLaunchingWithOptions:launchOptions]` in the Objective-C code.

File: platform-channels.md [hunk 4]
```diff
@@ -75,15 +75,22 @@ The following table shows how Dart values are received on the platform side and
...
+
+<br>
 ## Example: Calling platform-specific iOS and Android code using platform channels {#example}
 
 The following demonstrates how to call a platform-specific API to retrieve and
 display the current battery level. It uses the Android `BatteryManager` API, and
 the iOS `device.batteryLevel` API, via a single platform message,
 `getBatteryLevel`.
 
+The example adds the platform-specific code inside the main app itself. If you
```
Comment (mit-mit) [inline platform-channels.md]: Yes, should only mention once :-)

File: platform-channels.md [hunk 5]
```diff
@@ -298,10 +312,123 @@ You should now be able to run the app on Android. If you are using the Android
 Emulator, you can set the battery level in the Extended Controls panel
 accessible from the `...` button in the toolbar.
 
-### Step 3b: Add an iOS platform-specific implementation using Objective-C {#example-objc}
+### Step 3b: Add an Android platform-specific implementation using Kotlin {#example-android-kotlin}
```
Comment (mit-mit) [inline platform-channels.md]: Good point, fixed!

File: platform-channels.md [hunk 6]
```diff
@@ -298,10 +312,123 @@ You should now be able to run the app on Android. If you are using the Android
...
+Start by opening the Android host portion of your Flutter app in Android Studio:
+
+1. Start Android Studio
+
+1. Select the menu item 'File > Open...'
+
+1. Navigate to the directory holding your Flutter app, and select the `android`
+folder inside it. Click OK.
+
+1. Open the file `MainActivity.kt` located in the `java` folder in the Project
```
Comment (mit-mit) [inline platform-channels.md]: Good catch

File: platform-channels.md [hunk 7]
```diff
@@ -298,10 +312,123 @@ You should now be able to run the app on Android. If you are using the Android
...
+Next, we add the actual Android Kotlin code that uses the Android battery APIs to
+retrieve the battery level. This code is exactly the same as you would have
+written in a native Android app.
+
+Add the following as a new method in the `MainActivity` class, below the `onCreate`
+method:
+
+```kotlin
+  private fun getBatteryLevel(): Int {
+    var batteryLevel = -1
```
Comment (mit-mit) [inline platform-channels.md]: 👍

File: platform-channels.md (hunk 3 above)
Comment (mit-mit) [inline platform-channels.md]: Added

**Pull request merged**